sycl/test/check_device_code/cuda/matrix/matrix-nvptx-tf32-test.cpp

```diff
@@ -79,15 +79,6 @@ int main() {
               sg, sub_c, accC.template get_multi_ptr<access::decorated::yes>(),
               N, layout::row_major);
 
-          // Round a, b to tf32
-          for (auto i = 0; i < 4; ++i)
-            get_wi_data(sg, sub_a)[i] =
-                round_to_tf32(get_wi_data(sg, sub_a)[i]);
```

Comment:
I've just realized that to be consistent `round_to_tf32` should also have a device code check. I removed it here since it didn't have one, but it will be better to add it back via
```
                auto round_lambda = [](auto &x) { x = round_to_tf32(x); };
                joint_matrix_apply(sg, sub_a, round_lambda);
```

and check it calls the correct nvvm builtin.

Reply:
Done this